app/routes.py

```diff
@@ -0,0 +1,302 @@
+                        } for video in videos]
+        return jsonify(videos_list)
+    elif request.method == 'POST':
+        request_body = request.get_json()
+        if 'title' not in request_body.keys():
+            return make_response({"details": "Request body must include title." }, 400)
+        elif 'release_date' not in request_body.keys():
+            return make_response({"details":"Request body must include release_date."}, 400)
+        elif 'total_inventory' not in request_body.keys():
+            return make_response({"details": "Request body must include total_inventory."}, 400)
+        else:
+            new_video = Video(title=request_body["title"], 
+                                release_date=request_body["release_date"],
+                                total_inventory=request_body["total_inventory"])
+
+            db.session.add(new_video)
+            db.session.commit()
+
+            return make_response({"id": new_video.id,
+                                    "title": new_video.title,
+                                    "release_date": new_video.release_date,
+                                    "total_inventory": new_video.total_inventory
+                                                }, 201)
+
+@video_bp.route('/<video_id>', methods=['GET', 'PUT', 'DELETE'])
+def handle_one_video(video_id):
+    # Even though the end-user may type a number as an argument for video_id, its data type isn't automatically int.
+    # The try/except clause coerces the typed argument to be an int. If it is a non-numeric value, then a ValueError will be raised.
+    # Alternatively, can also use isnumeric(). This will return a boolean, so no need for try/except. An if/else clause will suffice.
+    try:
+        id = int(video_id)
+    except ValueError:
+        return jsonify({"details": "Invalid data"}), 400
+
+    video = Video.query.get(video_id)
+
+    if video is None:
+        return jsonify({"message": f"Video {video_id} was not found"}), 404
+
+    one_video = {"title": video.title,
+                "id": video.id,
+                "total_inventory": video.total_inventory}
+
+    if request.method == 'GET':
+        return one_video
+    elif request.method == 'PUT':
+        updates = request.get_json()
+        if 'title' not in updates.keys() or 'release_date' not in updates.keys() \
+                                         or 'total_inventory' not in updates.keys():
+            return make_response({"details": "Invalid data"}, 400)
+        else:
+            video.title = updates['title']
+            video.release_date = updates['release_date']
+            video.total_inventory = updates['total_inventory']
+
+            db.session.commit()
+            return make_response({"id": video.id,
+                                    "title": video.title,
+                                    "release_date": video.release_date,
+                                    "total_inventory":video.total_inventory}) 
```

Comment:
this repeated code could be moved to your Video Model as an instance method or moved to a helper function. It could look something like after the instance method is created
```suggestion
            return make_response({
                        video.video_to_dict()
            }) 
```